### PR TITLE
swift 4 (not to be merged)

### PR DESCRIPTION
**Pull request by yarneo**

### CODE REVIEW

File: Mixpanel/Variant.swift [hunk 1]
```diff
@@ -157,7 +157,7 @@ class Variant: NSObject, NSCoding {
         if object === self {
             return true
         } else {
-            return self.ID == object.ID
+            return self.ID == object.ID && self.actions == object.actions
```
Comment (pchien) [inline Mixpanel/Variant.swift]: it looks like VariantAction's equal check only checks the name. is that enough to check if they're actually doing the same thing? i.e. if someone changes an action in their ab test will the name that's sent back change too?

also do we want to do this on a separate branch to fix it on master earlier than september?

**Comment by matt-loflin** [comment]
Proposed fix for Xcode 9 Beta 3 error:
Add Objective-C name to line 117 of TweakPersistency.swift
```@objc(TweakDiskPersistencyData) private final class Data: NSObject, NSCoding {```

File: Mixpanel/Variant.swift (hunk 1 above)
Comment (yarneo) [inline Mixpanel/Variant.swift]: it actually does change the name of the action after every change!

Comment (yarneo) [inline Mixpanel/Variant.swift]: and i will push out a fix earlier to master for this

**Comment by matt-loflin** [comment]
@yarneo Any chance you could merge #105 into this branch?

**Pull request closed without merging**